zds/utils/api/serializers.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth.models import AnonymousUser
+from rest_framework.serializers import ModelSerializer, SerializerMethodField, IntegerField, ChoiceField
+from zds.utils.models import Comment
+from zds.member.api.serializers import UserListSerializer
+
+
+class LikesSerializer(ModelSerializer):
+    count = IntegerField(source='like', read_only=True)
+    list = UserListSerializer(source='get_likers', many=True, read_only=True)
+
+    class Meta:
+        model = Comment
+        fields = ('count', 'list',)
+
+
+class DislikesSerializer(LikesSerializer):
+    count = IntegerField(source='dislike', read_only=True)
+    list = UserListSerializer(source='get_dislikers', many=True, read_only=True)
+
+
+class KarmaSerializer(ModelSerializer):
+    like = LikesSerializer(source='*', read_only=True)
+    dislike = DislikesSerializer(source='*', read_only=True)
+    user = SerializerMethodField()
+    vote = ChoiceField(choices=['like', 'dislike', 'neutral'], write_only=True)
+
+    class Meta:
+        model = Comment
+        fields = ('like', 'dislike', 'user', 'vote',)
+
+    def get_user(self, obj):
+        request = self.context.get('request', None)
+        if request is not None:
+            user = request.user
+        else:
```

Comment:
ouaip, c'est `if request.user` qu'il faut mettre.

Le 28/03/2016 11:24, Gérard Paligot a écrit :
>
> In zds/utils/api/serializers.py 
> <https://github.com/zestedesavoir/zds-site/pull/3265#discussion_r57557203>:
>
> > +
> > +class KarmaSerializer(ModelSerializer):
> > +    like = LikesSerializer(source='*', read_only=True)
> > +    dislike = DislikesSerializer(source='*', read_only=True)
> > +    user = SerializerMethodField()
> > +    vote = ChoiceField(choices=['like', 'dislike', 'neutral'], write_only=True)
> > +
> > +    class Meta:
> > +        model = Comment
> > +        fields = ('like', 'dislike', 'user', 'vote',)
> > +
> > +    def get_user(self, obj):
> > +        request = self.context.get('request', None)
> > +        if request is not None:
> > +            user = request.user
> > +        else:
>
> Dans quel cas cela peut arriver ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/zestedesavoir/zds-site/pull/3265/files/f896364cc5fa324a343a91e6861e7050ce7e3ae6#r57557203>
>